### PR TITLE
Added Panoptic Segmentation guide

### PR DESCRIPTION
**Pull request by kapil-varshney**
Adds guide for Panoptic Segmentation using MaXDeepLab model.

### CODE REVIEW

**Comment by kapil-varshney** [comment]
@BP-Ent Can you please review this guide, here? Please ignore the previous PR.

File: guide/14-deep-learning/panoptic_segmentation_with_maxdeeplab.ipynb [hunk 1]
```diff
@@ -0,0 +1,189 @@
+{
```
Comment (BP-Ent) [inline guide/14-deep-learning/panoptic_segmentation_with_maxdeeplab.ipynb:1]: <p>Panoptic segmentation [1] is a computer vision task that creates a complete segmentation of an image and labes each pixel with a class label, including both "stuff" classes (e.g., sky, road, grass) and "thing" classes (e.g., person, car, tree). In other words, panoptic segmentation aims to perform both instance segmentation (segmenting each object instance as a separate entity) and semantic segmentation (labeling each pixel with a class label) in a single model. </p><p>The goal of panoptic segmentation is to provide a more complete and accurate representation of an image by not only detecting and labeling individual objects, but also understanding the relationships between objects and the scene as a whole. This can be useful for a variety of applications, such as remote sensing, autonomous driving, robotics, and augmented reality. </p><p>Panoptic segmentation models are typically trained on large datasets of annotated images and use techniques from deep learning, such as convolutional neural networks (CNNs), and recently transformers, to learn to segment images. The performance of panoptic segmentation algorithms is typically evaluated using the panoptic quality (PQ) metric, which measures the overall quality of the predicted segmentation in terms of its completeness and consistency.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1431/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (BP-Ent) [inline guide/14-deep-learning/panoptic_segmentation_with_maxdeeplab.ipynb:1]: <p>MaXDeepLab is the "first end-to-end model for panoptic segmentation" that predicts a set of non-overlapping masks and their classes directly [2]. While earlier panoptic segmentation models use hand-coded priors, such as anchors for objects, and other techniques, such as instance and semantic segmentation merging [3], MaXDeepLab combines these multiple surrogate tasks and provides a unified model. MaXDeepLab architecture includes a dual-path transformer, a stacked decoder, and two output heads that predict the masks and the classes.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1431/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (BP-Ent) [inline guide/14-deep-learning/panoptic_segmentation_with_maxdeeplab.ipynb:1]: <p>A transformer [4] is a type of neural network architecture. It is primarily used for natural language processing (NLP) tasks, such as language translations, but is increasingly making its way into computer vision. The main innovation of the transformer architecture is the use of self-attention mechanisms that allow the model to weight input features differently at different positions in the input sequence. This helps the model to understand long-range dependencies in the input data, as the self-attention mechanism allows the model to attend to relevant features at any position in the input sequence. </p><p>MaXDeepLab also uses the axial attention block from the Axial-DeepLab [5] model for its pixel-to-pixel self attention blocks. Axial attention blocks use two 1D attentions along height and width, sequentially. This creates position sensitive attentions, while keeping the model computationally less expensive. This is in contrast to the 2D attentions that have become common with image inputs and that introduce either computational complexity or localization constraints. Axial attention efficiently attends to a large receptive field. The MaX-Deeplab name is derived from extending the Axial-Deeplab with a <strong>Ma</strong>sk <strong>X</strong>former [2].</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1431/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Comment (BP-Ent) [inline guide/14-deep-learning/panoptic_segmentation_with_maxdeeplab.ipynb:1]: <p>MaXDeepLab employs CNNs along with transformers in a novel architecture called the dual-path transformer block. The CNNs and the transformer are integrated in a dual-path fashion, allowing bi-directional communication amongst the pixels and the global memory paths. This enables the CNNs to read and write from the global memory at any layer [2]. This contributes to a better performing model. </p><p>Another important characteristic of MaXDeepLab is the use of a loss function that is inspired by Panoptic Quality (PQ), the evaluation metric for panoptic segmentation. Panoptic quality is defined as the multiplication of recognition quality and segmentation quality [2]. Recognition quality represents the similarity between the classes and segmentation quality represents the similarity between the masks.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1431/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Comment (BP-Ent) [inline guide/14-deep-learning/panoptic_segmentation_with_maxdeeplab.ipynb:1]: <p>Here, <code>data</code> is the databunch prepared for training using the <code>prepare_data</code> method in the earlier steps. </p><p>For more information about the API, please see the <a href="https://developers.arcgis.com/python/api-reference/arcgis.learn.toc.html#maxdeeplab" target="_blank">API reference</a>.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1431/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

**Review by BP-Ent — CHANGES_REQUESTED** [review]
Suggested changes made on reviewnb

**Comment by priyankatuteja** [comment]
@kapil-varshney please incorporate the changes

**Comment by kapil-varshney** [comment]
@priyankatuteja Done.

File: guide/14-deep-learning/panoptic_segmentation_with_maxdeeplab.ipynb (hunk 1 above)
Comment (priyankatuteja) [inline guide/14-deep-learning/panoptic_segmentation_with_maxdeeplab.ipynb:1]: <p>fix formatting issue in the last two words</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1431/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Comment (priyankatuteja) [inline guide/14-deep-learning/panoptic_segmentation_with_maxdeeplab.ipynb:1]: <p>put model name in backticks <code>MaXDeepLab</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1431/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (priyankatuteja) [inline guide/14-deep-learning/panoptic_segmentation_with_maxdeeplab.ipynb:1]: <p>remove full stop from the first bullet</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1431/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Comment (kapil-varshney) [reply guide/14-deep-learning/panoptic_segmentation_with_maxdeeplab.ipynb:1]: Done.

Comment (kapil-varshney) [reply guide/14-deep-learning/panoptic_segmentation_with_maxdeeplab.ipynb:1]: Putting the model name in backticks shouldn't be necessary here as we are not saying anything about the code or the API.

Comment (kapil-varshney) [reply guide/14-deep-learning/panoptic_segmentation_with_maxdeeplab.ipynb:1]: That is done deliberately to highlight the name of the model.

**Review by priyankatuteja — APPROVED** [review]
